lib/matplotlib/backends/backend_wx.py

```diff
@@ -1033,11 +1041,56 @@ def _onEnter(self, evt):
         evt.Skip()
         FigureCanvasBase.enter_notify_event(self, guiEvent=evt, xy=(x, y))
 
+    def _draw_rubberband(self, x0, y0, x1, y1):
+        # trigger a refresh to draw a rubberband-like selection box
+        previous_rubberband = self._rubberband
+        height = self.figure.bbox.height
+        y1 = height - y1
+        y0 = height - y0
+        self._rubberband = [(x0, y0), (x1, y1)]
+        self._refresh_rubberband(previous_rubberband)
+
+    def _remove_rubberband(self):
+        # end drawing of a rubberband-like selection box
+        if not self._rubberband:
+            return
+        self._overlay.Reset()
+        self._overlay = None
+        self._refresh_rubberband()  # trigger a later redraw
+        self._rubberband = None
+
+    def _refresh_rubberband(self, previous_rubberband=None):
+        # initiate a refresh on the area that contains the previous and
+        # current selection / rubberband
+        points = list(self._rubberband)
+        if previous_rubberband:
+            points += previous_rubberband
+        # find the rectangle that covers the previous and current selection
+        top = min(point[1] for point in points)
+        left = min(point[0] for point in points)
+        bottom = max(point[1] for point in points)
+        right = max(point[0] for point in points)
+
+        if previous_rubberband:
+            # extend by one pixel to avoid residuals on Mac OS
+            if left > 0:
```

Comment:
I tend to write these `left = max(left - 1, 0)` as the bound is clearer (you don't need to check whether it was > or >=).  Not that it really matters here, just a suggestion.